#To-Do List

###General
- [x] Organise GitHub repositoryby adding new subdirectories and rearranging files (sequences, biographies, tutorials)
- [x] Create list of meetings and a to-do list
- [x] Update README.md file

===

###Weekly Goals
#### General
- [ ] Python tutorials!
- [ ] Git

####Simon and Alan
- [ ] script to read EMBOSS output and present the restriction maps
  - [ ] correct gaps so that each restriction enzyme site refers to a colun in the aligned sequence
  - [ ] write a general script

####Maxim and Imogen
- [ ] realign .fasta output in Jalview, upload the file
- [ ] script to find restriction enzymes sites to distinguish two species

===

###Planning Ahead

- [ ] think about Primer3 input and output

===

#Meetings
###Meeting 1 - Monday 25/Jan
####Overview
1. What has been achieved
  - ___GiHub account___ set up and ___Git tutorial___ completed
  - Progress through ___Python tutorials___, but _not completed_
  - Ability to ___orientate in NCBI___, and fetch the correct and complete nucleotide sequences
  - Capable of ___working on Jalview___, and aligning and editing sequences on display
2. Current state of the project
  - Table of contents for species completed and displayed on ___README.md___
  - NCBI sequences aligned and processed
3. Plans and discussion
The next step is to shift from Jalview to Python, by processing the raw data from Jalview into Python;
Sort the following by ___strings and loops____ on Python
  - Sort sequences and gaps
  - Realign circular mitochondrial sequences

===

####Roles and Attendance
| Group Member   | Roles         | Attendance  |
| -------------  |:-------------:| -----------:|
| Simon Bajew    | Presenter     | Present     |
| Imogen Stafford| Chair         | Present     |
| Maxim Tsenkov  | Minute Taker  | Present     |
| Alan Keith     |       -       | Absent      |
